src/main/java/org/fogbowcloud/app/jes/arrebol/ArrebolJobSynchronizer.java

```diff
@@ -24,70 +23,73 @@ public ArrebolJobSynchronizer(Properties properties) {
-		while (keys.hasNext()) {
-			String key = keys.next();
-			JSONObject taskObject = jsonObject.getJSONObject(key);
-			String taskState = taskObject.getString("state");
-			LOGGER.debug("State [" + taskState + "] of [" + key + "] Task");
-			taskStateList.add(ArrebolTaskState.getTaskStateFromDesc(taskState));
-		}
-		return taskStateList;
+	private void updateJob(JDFJob job, ArrebolJob arrebolJob){
+		updateTasksState(job, arrebolJob);
+		updateJobState(job, arrebolJob.getJobState());
 	}
-	
-	public JDFJobState getJobState(Set<ArrebolTaskState> taskStates) {
-		if (taskStates.contains(ArrebolTaskState.FAILED)) {
-			return JDFJobState.FAILED;
-		} else if (taskStates.contains(ArrebolTaskState.RUNNING)) {
-			return JDFJobState.SUBMITTED;
-		} else if (taskStates.contains(ArrebolTaskState.FINISHED)) {
-			if(taskStates.contains(ArrebolTaskState.PENDING)) {
-				return JDFJobState.SUBMITTED;
-			} else {
-				return JDFJobState.FINISHED;
-			}
-		} else {
-			return JDFJobState.CREATED;
+
+	private void updateTasksState(JDFJob job, ArrebolJob arrebolJob){
+		Map<String, ArrebolTask> arrebolTasks = arrebolJob.getTasks();
+		LOGGER.info("Updating tasks state from job [" + job.getId() + "].");
+		for(Task task : job.getTaskList().values()){
+			ArrebolTask arrebolTask = arrebolTasks.get(task.getId());
+			ArrebolTaskState arrebolTaskState = arrebolTask.getState();
+			TaskState taskState = getTaskState(arrebolTaskState);
+			task.setState(taskState);
+			LOGGER.debug("Updated task [" + task.getId() + "] to state " + taskState.toString());
 		}
 	}
 
-	public enum ArrebolTaskState {
-		PENDING("PENDING"), FAILED("FAILED"), FINISHED("FINISHED"), RUNNING("RUNNING"), CLOSED("CLOSED");
-
-		private String desc;
-
-		ArrebolTaskState(String desc) {
-			this.desc = desc;
-		}
+	private void updateJobState(JDFJob job, ArrebolJobState arrebolJobState){
+		JDFJobState jdfJobState = this.getJobState(arrebolJobState);
+		job.setState(jdfJobState);
+		LOGGER.info("Updated job [" + job.getId() + "] to state " + jdfJobState.toString());
+	}
 
-		public String getDesc() {
-			return this.desc;
+	private TaskState getTaskState(ArrebolTaskState arrebolTaskState){
+		if(arrebolTaskState.equals(ArrebolTaskState.FAILED)){
+			return TaskState.FAILED;
+		} else if(arrebolTaskState.equals(ArrebolTaskState.RUNNING)){
+			return TaskState.RUNNING;
+		} else if(arrebolTaskState.equals(ArrebolTaskState.FINISHED)){
+			return TaskState.FINISHED;
+		} else if(arrebolTaskState.equals(ArrebolTaskState.PENDING)){
+			return TaskState.READY;
+		} else if(arrebolTaskState.equals(ArrebolTaskState.CLOSED)){
+			return TaskState.COMPLETED;
 		}
+		return null;
+	}
 
-		public static ArrebolTaskState getTaskStateFromDesc(String desc) {
-			for (ArrebolTaskState ts : values()) {
-				if (ts.getDesc().equals(desc)) {
-					return ts;
-				}
-			}
-			return null;
+	private JDFJobState getJobState(ArrebolJobState arrebolJobState){
```

Comment:
switch?

Reply:
Done.

Reply:
did you miss your commit? where is the switch?